src/main/scala/chisel3/util/ExtModuleUtils.scala

```diff
@@ -18,7 +19,13 @@ trait HasExtModuleResource extends ExtModule {
     */
   def addResource(blackBoxResource: String): Unit = {
     val anno = new ChiselAnnotation with RunFirrtlTransform {
-      def toFirrtl = BlackBoxResourceAnno(self.toNamed, blackBoxResource)
+      def toFirrtl = try {
```

Comment:
Quibble:  Perhaps this code could be DRYed out a bit by merging the identical code between BlackBox and ExtModule into a trait

Reply:
Done.